pkg/security/probe/probe_bpf.go

```diff
@@ -473,11 +451,24 @@ func (p *Probe) handleEvent(CPU int, data []byte, perfMap *manager.PerfMap, mana
 			log.Errorf("failed to decode removexattr event: %s (offset %d, len %d)", err, offset, len(data))
 			return
 		}
+	case ExecEventType, ForkEventType:
+		if _, err := event.Exec.UnmarshalEvent(data[offset:], event); err != nil {
+			log.Errorf("failed to decode exec event: %s (offset %d, len %d)", err, offset, len(data))
+			return
+		}
+
+		// update the process resolver cache
+		event.updateProcessCachePointer(p.resolvers.ProcessResolver.AddEntry(event.Process.Pid, event.processCacheEntry))
+	case ExitEventType:
+		defer p.resolvers.ProcessResolver.DeleteEntry(event.Process.Pid, event.ResolveEventTimestamp())
```

Comment:
should we return here to avoid useless dispatch and cache resolution ?

Reply:
I'm not sure there is a cache resolution (I mean other than the ProcessCacheEntry resolution which will happen anyway because of the `DeleteEntry`), also I'm afraid the tests won't work anymore if we don't dispatch the event :(